# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1553006812 0
#      Tue Mar 19 14:46:52 2019 +0000
# Node ID 6acb713f823193049dffc8935b5250e7f288ca6f
# Parent  58e3009545180ccb7d6fe0fb2fbbf5e86a0427ac
avoid conflict with @seealso macro in Texinfo 6.6 (bug #55952)

* macros.texi: Define @xseealso instead of @seealso.

* munge-texi.pl: Translate @seealso in DOCSTRINGS and source files to
@xseealso.

diff --git a/doc/interpreter/macros.texi b/doc/interpreter/macros.texi
--- a/doc/interpreter/macros.texi
+++ b/doc/interpreter/macros.texi
@@ -41,19 +41,24 @@
 @c The following macro is used for the on-line help system, but we don't
 @c want lots of `See also: foo, bar, and baz' strings cluttering the
 @c printed manual (that information should be in the supporting text for
 @c each group of functions and variables).
 @c
 @c Implementation Note:
 @c For TeX, @vskip produces a nice separation.
 @c For Texinfo, '@sp 1' should work, but in practice produces ugly results
-@c for HTML.  We use a simple blank line to produce the correct behavior.
+@c for HTML.  We use a simple blank line to produce the correct
+@c behavior.
+@c
+@c We use @xseealso now because Texinfo introduced its own @seealso
+@c command.  But instead of modifying all source files, we'll have the
+@c munge-texi script convert @seealso to @xseealso.
 
-@macro seealso {args}
+@macro xseealso {args}
 @iftex
 @vskip 2pt
 @end iftex
 @ifnottex
 
 @end ifnottex
 @ifnotinfo
 @noindent
diff --git a/doc/interpreter/munge-texi.pl b/doc/interpreter/munge-texi.pl
--- a/doc/interpreter/munge-texi.pl
+++ b/doc/interpreter/munge-texi.pl
@@ -68,16 +68,18 @@ foreach $DOCSTRING_file (@ARGV)
 ################################################################################
 # Process .txi to .texi by expanding @DOCSTRING, @EXAMPLEFILE macros
 
 # Add warning header
 print '@c DO NOT EDIT!  Generated automatically by munge-texi.pl.',"\n\n";
 
 TXI_LINE: while (<STDIN>)
 {
+  s/\@seealso/\@xseealso/g;
+
   if (/^\s*\@DOCSTRING\((\S+)\)/)
   {
     $func = $1;
     $docstring = $help_text{$func};
     if (! $docstring)
     {
       warn "warning: no DOCSTRING entry for $func\n";
       next TXI_LINE;
@@ -134,16 +136,18 @@ sub extract_docstring
       foreach $func (split (/,/, $func_list))
       {
         $func =~ s/^@/@@/;   # Texinfo uses @@ to produce '@'
         $node = $func;
         $node =~ s/\./_/g;  # Texinfo doesn't like '.' in node names
         $repl .= "\@ref{XREF$node,,$func}, ";
       }
       substr($repl,-2) = "";   # Remove last ', '
-      $_ = "\@seealso{$repl}$rest_of_line";
+      # write out @xseealso because we have our own macro that conflicts
+      # with the one introduced in Texinfo 6.
+      $_ = "\@xseealso{$repl}$rest_of_line";
     }
 
     $docstring .= $_;
   }
 
   return $docstring;
 }
